drivers/sensors/apds9922.c

```diff
@@ -0,0 +1,2598 @@
+ *  priv   - pointer to device structure
+ *  factor - als factor to use
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_als_factor(FAR struct apds9922_dev_s *priv,
+                               uint32_t factor)
+{
+  if (factor < 1)
+    {
+      return -EINVAL;
+    }
+
+  priv->als_setup.als_factor = factor;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_limit
+ *
+ * Description:
+ *   Sets the ALS auto range limit - "limit percent" of full scale value
+ *
+ * Input Parameters:
+ *  priv - pointer to device structure
+ *  limi - limit to use (1-100 %)
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_als_limit(FAR struct apds9922_dev_s *priv,
+                              uint32_t limit)
+{
+  if ((limit < 1) || (limit > 100))
+    {
+      return -EINVAL;
+    }
+
+  priv->als_setup.range_lim = limit;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_int_mode
+ *
+ * Description:
+ *   Sets the ALS interrupt mode - disabled, threshold or variance.
+ *
+ * Input Parameters:
+ *  priv    - pointer to device structure
+ *  channel - interrupt mode
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_als_int_mode(FAR struct apds9922_dev_s *priv,
+                                       enum apds9922_als_int_mode mode)
+{
+  uint8_t regval;
+  int ret;
+
+  if (mode > ALS_INT_MODE_VARIANCE)
+    {
+      return -EINVAL;
+    }
+
+  ret = apds9922_i2c_read8(priv, APDS9922_INT_CFG, &regval);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  regval &= ~ALS_INT_MASK;
+
+  switch (mode)
+    {
+      case ALS_INT_MODE_VARIANCE:
+        regval |= ALS_INT_VAR_MODE | ALS_INT_EN;
+        break;
+      case ALS_INT_MODE_THRESHOLD:
+        regval |= ALS_INT_EN;
+        break;
+      case ALS_INT_MODE_DISABLED:
+      default:
+        break;
+    }
+
+  ret = apds9922_i2c_write8(priv, APDS9922_INT_CFG, regval);
+
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  priv->als_setup.int_mode = mode;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_thresh
+ *
+ * Description:
+ *   Sets the ALS thresholds, upper and lower.
+ *
+ * Input Parameters:
+ *  priv       - pointer to device structure
+ *  thresholds - struct of thresholds to set
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_als_thresh(FAR struct apds9922_dev_s *priv,
+                                       FAR struct adps9922_als_thresh
```

Comment:
```suggestion
                                       struct adps9922_als_thresh
```
Since it is not a pointer.
BTW, wouldn't be usage of a const pointer more efficient compared to copy args?